src/wasm.rs

```diff
@@ -27,24 +32,29 @@ const SECTION_WC_JWT: &str = "wasmcloud_jwt";
 pub fn extract_claims(contents: impl AsRef<[u8]>) -> Result<Option<Token<Actor>>> {
     let target_hash = compute_hash(&strip_custom_section(contents.as_ref())?)?;
     let parser = wasmparser::Parser::new(0);
+    let mut depth = 0;
     for payload in parser.parse_all(contents.as_ref()) {
-        match payload? {
-            wasmparser::Payload::CustomSection(reader) => {
-                if reader.name() == SECTION_JWT || reader.name() == SECTION_WC_JWT {
-                    let jwt = String::from_utf8(reader.data().to_vec())?;
-                    let claims: Claims<Actor> = Claims::decode(&jwt)?;
-                    if let Some(ref meta) = claims.metadata {
-                        if meta.module_hash != target_hash
-                            && claims.wascap_revision.unwrap_or_default()
-                                >= MIN_WASCAP_INTERNAL_REVISION
-                        {
-                            return Err(errors::new(ErrorKind::InvalidModuleHash));
-                        } else {
-                            return Ok(Some(Token { jwt, claims }));
-                        }
+        let payload = payload?;
+        use wasmparser::Payload::*;
+        match payload {
+            ModuleSection { .. } | ComponentSection { .. } => depth += 1,
+            End { .. } => depth -= 1,
+            CustomSection(c)
+                if (c.name() == SECTION_JWT) || (c.name() == SECTION_WC_JWT) && depth == 0 =>
+            {
+                let jwt = String::from_utf8(c.data().to_vec())?;
+                let claims: Claims<Actor> = Claims::decode(&jwt)?;
+                if let Some(ref meta) = claims.metadata {
+                    if meta.module_hash != target_hash
+                        && claims.wascap_revision.unwrap_or_default()
+                            >= MIN_WASCAP_INTERNAL_REVISION
```

Comment:
should this be 
```
 if meta.module_hash != target_hash
                        || claims.wascap_revision.unwrap_or_default()
                            < MIN_WASCAP_INTERNAL_REVISION
```
(changed `&&` to `||` and `>=` to `<`)

Reply:
no, this is the same logic that was there before. If the hashes don't match and the wascap revision is "the new one", then we proceed with the check. If you look at the diff you'll see that this logic didn't change in the PR.

Reply:
Sanity check: I agree the logic didn't change here but would like to make sure it's what we want.

If hashes don't match AND this is a module that SHOULD match aka it meets the minimum wascap version that it should be supported, then we should ERROR. => LGTM

If the hashes don't match and the module is on an older internal revision, then it might not be a problem because we don't know how it used to compute the hash? So then we continue on with the claims in the module? Is this correct?